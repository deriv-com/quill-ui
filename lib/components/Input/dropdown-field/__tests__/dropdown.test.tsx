import React from "react";
import { render, fireEvent, screen, act } from "@testing-library/react";
import InputDropdown from "../index";

// Mock options for dropdown
const options = [
    { value: "1", text: "Option 1" },
    { value: "2", text: "Option 2" },
    { value: "3", text: "Option 3" },
];

describe("InputDropdown component", () => {
    it("renders dropdown with options", () => {
        const dropdownProps = {
            options: options,
            onSelectOption: (value: string) => {
                return value;
            },
        };
        render(<InputDropdown {...dropdownProps} />);
        act(() => {
            fireEvent.click(screen.getByTestId("dropdown-input"));
        });

        options.forEach((option) => {
            expect(screen.getByText(option.text)).toBeInTheDocument();
        });
    });

    it("selects an option from dropdown", () => {
        const dropdownProps = {
            options: options,
            onSelectOption: (value: string) => {
                return value;
            },
        };
        render(<InputDropdown {...dropdownProps} />);
        act(() => {
            fireEvent.click(screen.getByTestId("dropdown-input"));
        });
        act(() => {
            fireEvent.click(screen.getByText(options[0].text));
        });

        expect(screen.getByTestId("dropdown-input")).toHaveValue(
            options[0].text,
        );
    });

    it("closes dropdown on selection", () => {
        const dropdownProps = {
            options: options,
            onSelectOption: (value: string) => {
                return value;
            },
        };
        render(<InputDropdown {...dropdownProps} />);

        act(() => {
            fireEvent.click(screen.getByTestId("dropdown-input"));
        });
        act(() => {
            fireEvent.click(screen.getByText(options[0].text));
        });

        expect(screen.queryByText(options[0].text)).not.toBeInTheDocument();
    });
    it("should render a TextField with success status", () => {
        const dropdownProps = {
            options: options,
            onSelectOption: (value: string) => {
                return value;
            },
        };
        const { container } = render(
            <InputDropdown {...dropdownProps} status="success" />,
        );
        expect(container).toMatchSnapshot();
    });

    it("should render a TextField with error status", () => {
        const dropdownProps = {
            options: options,
            onSelectOption: (value: string) => {
                return value;
            },
        };
        const { container } = render(
            <InputDropdown {...dropdownProps} status="error" />,
        );
        expect(container).toMatchSnapshot();
    });
});
