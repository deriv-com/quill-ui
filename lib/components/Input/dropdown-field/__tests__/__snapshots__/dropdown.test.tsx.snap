// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InputDropdown component should render a TextField with error status 1`] = `
<div>
  <div
    class="quill-custom-dropdown__container"
  >
    <div
      class="quill-custom-dropdown__head"
    >
      <div
        class="quill-input__container"
      >
        <div
          class="quill-input__wrapper quill-input__wrapper__variant--outline quill-input__wrapper__variant--outline--error quill-input__wrapper__size--lg"
        >
          <div
            class="quill-input-icons__wrapper"
          >
            <div
              class="quill-input-no-label__wrapper"
            >
              <input
                class="input peer input__align--left input__size--lg"
                data-testid="dropdown-input"
                id="quill-input-5"
                placeholder=" "
                readonly=""
                type="select"
                value=""
              />
            </div>
            <span
              class="icon_wrapper status-icon--error"
            >
              <svg
                height="24"
                role="img"
                viewBox="0 0 32 32"
                width="24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M15.688 9.82 7.953 22.516c-.078.117-.078.195-.078.312 0 .313.234.547.547.547h15.117c.313 0 .586-.234.586-.547 0-.117-.04-.195-.117-.312L16.273 9.82A.29.29 0 0 0 16 9.625c-.156 0-.234.078-.312.195m-1.602-.976A2.25 2.25 0 0 1 16 7.75c.742 0 1.484.43 1.875 1.094l7.734 12.695c.235.39.391.82.391 1.29 0 1.327-1.094 2.421-2.46 2.421H8.421A2.434 2.434 0 0 1 6 22.828q0-.703.352-1.289zm3.164 12.031c0 .703-.586 1.25-1.25 1.25-.703 0-1.25-.547-1.25-1.25 0-.664.547-1.25 1.25-1.25.664 0 1.25.586 1.25 1.25m-.312-7.187v3.75c0 .546-.43.937-.938.937-.547 0-.937-.39-.937-.937v-3.75c0-.508.39-.938.937-.938a.95.95 0 0 1 .938.938"
                />
              </svg>
            </span>
            <svg
              class="quill-input__rotate quill-input__icon"
              data-state="close"
              height="24"
              role="img"
              viewBox="0 0 14 22"
              width="24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <g>
                <path
                  d="m6.535 14.715-5.25-5.25c-.273-.246-.273-.656 0-.93.246-.246.656-.246.93 0L7 13.348l4.785-4.786c.246-.273.656-.273.93 0a.644.644 0 0 1 0 .903l-5.277 5.25a.6.6 0 0 1-.903 0"
                />
              </g>
              <defs>
                <clippath
                  id="b2fbae3ff4787741870694db3344c0c7__a"
                >
                  <path
                    d="M0 0h14v22H0z"
                  />
                </clippath>
              </defs>
            </svg>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`InputDropdown component should render a TextField with success status 1`] = `
<div>
  <div
    class="quill-custom-dropdown__container"
  >
    <div
      class="quill-custom-dropdown__head"
    >
      <div
        class="quill-input__container"
      >
        <div
          class="quill-input__wrapper quill-input__wrapper__variant--outline quill-input__wrapper__variant--outline--success quill-input__wrapper__size--lg"
        >
          <div
            class="quill-input-icons__wrapper"
          >
            <div
              class="quill-input-no-label__wrapper"
            >
              <input
                class="input peer input__align--left input__size--lg"
                data-testid="dropdown-input"
                id="quill-input-4"
                placeholder=" "
                readonly=""
                type="select"
                value=""
              />
            </div>
            <span
              class="icon_wrapper status-icon--success"
            >
              <svg
                height="24"
                role="img"
                viewBox="0 0 32 32"
                width="24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M16 8.375c-2.93 0-5.586 1.563-7.07 4.063-1.446 2.539-1.446 5.624 0 8.124A8.13 8.13 0 0 0 16 24.626c2.89 0 5.547-1.523 7.031-4.062 1.446-2.5 1.446-5.586 0-8.125-1.484-2.5-4.14-4.063-7.031-4.063M16 26.5c-3.594 0-6.875-1.875-8.672-5-1.797-3.086-1.797-6.875 0-10 1.797-3.086 5.078-5 8.672-5 3.555 0 6.836 1.914 8.633 5 1.797 3.125 1.797 6.914 0 10a9.93 9.93 0 0 1-8.633 5m4.414-11.836-5 5c-.39.39-.976.39-1.328 0l-2.5-2.5a.856.856 0 0 1 0-1.289c.351-.39.937-.39 1.328 0l1.836 1.836 4.336-4.336c.352-.39.937-.39 1.328 0a.92.92 0 0 1 0 1.29"
                />
              </svg>
            </span>
            <svg
              class="quill-input__rotate quill-input__icon"
              data-state="close"
              height="24"
              role="img"
              viewBox="0 0 14 22"
              width="24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <g>
                <path
                  d="m6.535 14.715-5.25-5.25c-.273-.246-.273-.656 0-.93.246-.246.656-.246.93 0L7 13.348l4.785-4.786c.246-.273.656-.273.93 0a.644.644 0 0 1 0 .903l-5.277 5.25a.6.6 0 0 1-.903 0"
                />
              </g>
              <defs>
                <clippath
                  id="b2fbae3ff4787741870694db3344c0c7__a"
                >
                  <path
                    d="M0 0h14v22H0z"
                  />
                </clippath>
              </defs>
            </svg>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
